src/modules/CourseDetail/index.tsx

```diff
@@ -29,10 +35,56 @@ import { courseTypeStringFromCourse } from './utils/courseTypeStringFromCourse'
 import { groupBy } from './utils/groupBy'
 import { parseVariablesFromQuery } from './utils/parseVariablesFromQuery'
 
-export function CourseDetailPage(props: { data: GetCourseResponse }) {
+const DynamicReviewProvider = dynamic(
+  async () =>
+    (
+      await import(
+        /* webpackChunkName: "ReviewProvider" */
+        './context/Review'
+      )
+    ).ReviewProvider,
+  { ssr: false }
+)
+
+const DynamicReviewList = dynamic(
+  async () =>
+    (
+      await import(
+        /* webpackChunkName: "ReviewList" */
+        './components/ReviewList'
+      )
+    ).ReviewList,
+  { ssr: false }
+)
+
+const DynamicReviewForm = dynamic(
+  async () =>
+    (
+      await import(
+        /* webpackChunkName: "ReviewForm" */
+        './components/ReviewForm'
+      )
+    ).ReviewForm,
+  { ssr: false }
+)
```

Comment:
Separately declaring dynamic imports like this will output 3 separate bundle. I recommend wrapping these three in one component and dynamic import that instead.

<img width="544" alt="Screen Shot 2564-12-12 at 00 44 05" src="https://user-images.githubusercontent.com/8080853/145686395-5f7a5c12-1216-4ae9-a964-766e034ffd52.png">

Reply:
Why 3 chunk files are not good?

Reply:
They require you to make 3 HTTP requests to load them. And the inner ones aren't loaded until the parent finishes loading, causing a waterfall.